internal/store/store.go

```diff
@@ -1041,15 +1039,27 @@ func mkObjFromGVK(gvk schema.GroupVersionKind) (runtime.Object, error) {
 	// ----------------------------------------------------------------------------
 	// Kubernetes Gateway APIs
 	// ----------------------------------------------------------------------------
-	case gatewayv1beta1.SchemeGroupVersion.WithKind("HTTPRoutes"):
+	case gatewayv1beta1.SchemeGroupVersion.WithKind("HTTPRoute"):
```

Comment:
I wonder why 

- do we need this change?
- did this work before?

Same goes for all the other Gateway API related changes below 🤔 👀

Reply:
Only `NewCacheStoresFromObjYAML` uses `mkObjFromGVK`. `NewCacheStoresFromObjYAML` was only used in its own tests which do not test every GVK we support in the parser. Now when I got it off the shelf to actually use it for something else, I noticed it doesn't work as expected thus the changes. 

I could add some unit tests to that, but it will always be a matter of updating them when we add a new supported kind so I'm not sure if it would help, WDYT?

Reply:
Ok got it. In this case let it be. We want to test the production code, not test code. If we're getting the correct, expected output based in inputs in tests then we're good.